### PR TITLE
Migrate test-environments to Prow

### PR DESCRIPTION
**Pull request by jakkab**
**Description**

Changes proposed in this pull request:

- Add a Prow job for the test-environments component

**Related issue(s)**

### CODE REVIEW

File: development/tools/jobs/kyma/test-environments_test.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package kyma
```
[the remark this answers is not in the document]
Comment (jakkab) [reply development/tools/jobs/kyma/test-environments_test.go]: Done!

File: development/tools/jobs/kyma/test-environments_test.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+	assert.Equal(t, 10, actualPresubmit.MaxConcurrency)
+	assert.True(t, actualPresubmit.SkipReport)
+	assert.True(t, actualPresubmit.Decorate)
+	assert.Equal(t, "github.com/kyma-project/kyma", actualPresubmit.PathAlias)
+	tester.AssertThatHasExtraRefTestInfra(t, actualPresubmit.JobBase.UtilityConfig)
+	tester.AssertThatHasPresets(t, actualPresubmit.JobBase, tester.PresetDindEnabled, tester.PresetDockerPushRepo, tester.PresetGcrPush, tester.PresetBuildPr)
+	tester.AssertThatJobRunIfChanged(t, actualPresubmit, "tests/test-environments/fix")
+	assert.Equal(t, "^tests/test-environments/", actualPresubmit.RunIfChanged)
+	assert.Equal(t, tester.ImageGolangBuildpackLatest, actualPresubmit.Spec.Containers[0].Image)
+	assert.Equal(t, "/home/prow/go/src/github.com/kyma-project/kyma/tests/test-environments", actualPresubmit.Spec.Containers[0].Args[0])
```
Comment (suleymanakbas91) [inline development/tools/jobs/kyma/test-environments_test.go]: Please change the order of command and args to make it same with the other tests

File: development/tools/jobs/kyma/test-environments_test.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	assert.Equal(t, []string{"master"}, actualPost.Branches)
+
+	assert.Equal(t, 10, actualPost.MaxConcurrency)
+	assert.True(t, actualPost.Decorate)
+	assert.Equal(t, "github.com/kyma-project/kyma", actualPost.PathAlias)
+	tester.AssertThatHasExtraRefTestInfra(t, actualPost.JobBase.UtilityConfig)
+	tester.AssertThatHasPresets(t, actualPost.JobBase, tester.PresetDindEnabled, tester.PresetDockerPushRepo, tester.PresetGcrPush, tester.PresetBuildMaster)
+	assert.Equal(t, tester.ImageGolangBuildpackLatest, actualPost.Spec.Containers[0].Image)
+	assert.Equal(t, "/home/prow/go/src/github.com/kyma-project/kyma/tests/test-environments", actualPost.Spec.Containers[0].Args[0])
+
```
Comment (suleymanakbas91) [inline development/tools/jobs/kyma/test-environments_test.go]: Please add test for command as well

File: development/tools/jobs/kyma/test-environments_test.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
+package kyma_test
```
Comment (suleymanakbas91) [inline development/tools/jobs/kyma/test-environments_test.go]: Change file name as `test_environments_test.go`

Comment (jakkab) [reply development/tools/jobs/kyma/test-environments_test.go]: Done!

File: development/tools/jobs/kyma/test-environments_test.go (hunk 2 above)
Comment (jakkab) [reply development/tools/jobs/kyma/test-environments_test.go]: order of args unified across Equal assertions.

File: development/tools/jobs/kyma/test-environments_test.go (hunk 3 above)
Comment (jakkab) [reply development/tools/jobs/kyma/test-environments_test.go]: test added!

**Pull request merged**